### DISCUSSION TITLE
Exporting fields without data in version control

### DISCUSSION THREAD

**Discussion by mhabala** [pocketbase/pocketbase#2561] [Q&A] [answered]
I'm utilizing version control and seeking a way to export fields without including the data. In other words, I want the repository to include the database tables but not their respective entries, enabling a fresh start when running the repository from scratch. How can I accomplish something like this?

**Comment by ganigeorgiev** [comment]
If you are using the prebuilt executable, automigataions are enabled by default so you should have a `pb_migrations` directory next to your `pb_data`. The migrations directory is safe to be commited and all the unapplied migrations from this directory will be automatically applied on `serve`.

Alternatively, you also have also the option to export the collections configuration as JSON from the _Admin UI > Settings > Export collections_ and then reimport it in the other environment (but the recommended approach is to use `pb_migrations`).

---

When using PocketBase as Go framework the situation is similar. If you haven't already you can check https://pocketbase.io/docs/migrations/.

**Comment by mhabala** [comment]
![image](https://github.com/pocketbase/pocketbase/assets/94061901/83e80c66-4bc6-4398-a6dd-06cc8d1d1786)
I'm using it as a framework. I can't see the pb_migrations directory, despite having created a new collection. When I turn it off and on, the migrations directory doesn't exist, but the collection data is still stored.

**Reply by ganigeorgiev** [reply, marked as answer]
If you are using PocketBase as framework the migrations usually will be stored under the `migrations` dir and they will be embedded as part of your final executable when building the application. For how to enable the migrations in Go, please check https://pocketbase.io/docs/migrations/#enable-go-migrations.

**Comment by mhabala** [comment]
Thanks, why is it disabled by default?

**Reply by ganigeorgiev** [reply]
Mainly because users need to specify from where to load the migrations (aka. the `import _ "yourpackage/migrations"` part) and eventually configure their behavior (eg. not everyone may want the `Automigrate` option or they may want to allow external migrations in JS, etc.).
